### DISCUSSION TITLE
Find decoding of RTL-SDR received data in rtl_433

### DISCUSSION THREAD

**Discussion by lwuethri** [merbanan/rtl_433#2271] [Q&A]
Hello,
I hope you are having a nice day:)
I have a question concerning the decoding of a sample sequence sent from a weather station and received with a RTL-SDR stick.

I was able to feed the received samples into a gnuradio flowgraph which generates an amplitude recording of the demodulated samples (see attached picture). At the same time, I see the corresponding temperature at the weather station as well as in the rtl_433 interface.

<img width="960" alt="2022-12-04_22 6_ch1" src="https://user-images.githubusercontent.com/120006384/206121100-008e0473-1b48-4151-aec9-e68ecdced84f.png">

Now I am asking myself, how I can get the actual decoding of the amplitude recording. Is there a way with rtl_433 to not only see the final decoded value of the temperature but also which decoding scheme was applied to the originally received samples in order to obtain exactly this value?

Many thanks in advance. I wish you a nice week.

**Comment by zuckschwerdt** [comment]
You should get a sample recording: https://triq.org/rtl_433/ANALYZE.html
Then demod by putting the sample on https://triq.org/pdv/
or demod to OOK and use that (`rtl_433 -w OOK:- samplefile.cu8` or `rtl_433 -w samplefile.ook samplefile.cu8`)
and inspect the pulse data.
Read more about what our slicers do here https://triq.org/rtl_433/PULSE_FORMATS.html
The decoding of that data can then be read in the source https://github.com/merbanan/rtl_433/tree/master/src/devices
